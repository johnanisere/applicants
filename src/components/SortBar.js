import React from "react";
import PropTypes from "prop-types";

import SortBox from "../blocks/sortBox";

const App = ({ value, list, onHandleChange, name }) => (
  <SortBox>
    <SortBox.Label>
      <span>Sort By: </span>
    </SortBox.Label>
    <SortBox.DropDown>
      <SortBox.Select value={value} onChange={onHandleChange} name={name}>
        <option value="">Sort ...</option>
        {list.map((value, key) => (
          <option value={value} key={key}>
            {value}
          </option>
        ))}
      </SortBox.Select>
    </SortBox.DropDown>
  </SortBox>
);

App.propTypes = {
  list: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onHandleChange: PropTypes.func.isRequired
};

App.defaultProps = {
  name: "",
  list: [],
  value: "",
  onHandleChange: () => {}
};

export default App;
