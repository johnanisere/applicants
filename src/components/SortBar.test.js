import React from "react";
import { shallow } from "enzyme";

import SortBar from "./SortBar";
import SortBox from "../blocks/sortBox";

it("SortBar component exists", () => {
  const wrapper = shallow(<SortBar />);
  expect(wrapper.exists()).toBe(true);
});

it("SortBar component renders", () => {
  const wrapper = shallow(<SortBar />);
  expect(wrapper).toMatchSnapshot();
});

it(`SortBar should contain "SortBox", "SortBox.Label", "SortBox.DropDown", "SortBox.Select"`, () => {
  const wrapper = shallow(<SortBar />);
  expect(wrapper.find(SortBox)).toHaveLength(1);
  expect(wrapper.find(SortBox.Label)).toHaveLength(1);
  expect(wrapper.find(SortBox.Select)).toHaveLength(1);
  expect(wrapper.find(SortBox.DropDown)).toHaveLength(1);
});
