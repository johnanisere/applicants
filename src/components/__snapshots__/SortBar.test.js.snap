// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar component renders 1`] = `ShallowWrapper {}`;
